src/TorchVision/AutoAugment.cs

```diff
@@ -0,0 +1,162 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using static TorchSharp.torch;
+using F = TorchSharp.torchvision.transforms.functional;
+
+#nullable enable
+namespace TorchSharp
+{
+    public static partial class torchvision
+    {
+        private static Tensor apply_op(
+                                Tensor img,
+                                string op_name,
+                                float magnitude,
+                                InterpolationMode interpolation,
+                                IList<float>? fill)
+        {
+            if (op_name == "ShearX") {
```

Comment:
Once you have switched to enums, a switch statement may be faster and easier to read.